resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/submit/ClientSuite.scala

```diff
@@ -191,25 +191,32 @@ class ClientSuite extends SparkFunSuite with BeforeAndAfter {
     assert(configMap.getData.get(SPARK_CONF_FILE_NAME).contains("conf2key=conf2value"))
   }
 
-  test("All files from SPARK_CONF_DIR, except templates and spark config " +
+  test("All files from SPARK_CONF_DIR, " +
+    "except templates, spark config, binary files and are within size limit, " +
     "should be populated to pod's configMap.") {
     def testSetup: (SparkConf, Seq[String]) = {
       val tempDir = Utils.createTempDir()
-      val sparkConf = new SparkConf(loadDefaults = false).setSparkHome(tempDir.getAbsolutePath)
+      val sparkConf = new SparkConf(loadDefaults = false)
+        .setSparkHome(tempDir.getAbsolutePath)
 
       val tempConfDir = new File(s"${tempDir.getAbsolutePath}/conf")
       tempConfDir.mkdir()
       // File names - which should not get mounted on the resultant config map.
       val filteredConfFileNames =
-        Set("spark-env.sh.template", "spark.properties", "spark-defaults.conf")
-      val confFileNames = for (i <- 1 to 5) yield s"testConf.$i" ++
+        Set("spark-env.sh.template", "spark.properties", "spark-defaults.conf",
+          "test.gz", "test2.jar", "non_utf8.txt")
```

Comment:
@ScrapCodes, quick question. Is it possible to have a list of allowed items only instead?

Reply:
That would disallow a user from supplying his library/framework specific configuration.

Reply:
For example, if a user needs to pass his hadoop's core-site.xml. or some other framework he is trying to use, has a configuration file and for that matter, he has a custom application which can be configured.